src/Gui/CommandStd.cpp

```diff
@@ -924,6 +926,79 @@ void StdCmdReloadStyleSheet::activated(int )
     Application::Instance->reloadStyleSheet();
 }
 
+// ==================================================================
+// Std_AnnotationLabel
+// ==================================================================
+DEF_STD_CMD_A(StdCmdAnnotationLabel)
+
+StdCmdAnnotationLabel::StdCmdAnnotationLabel()
+  : Command("Std_AnnotationLabel")
+{
+    // Put it under Tools menu
+    sGroup        = QT_TR_NOOP("Tools");
+    sMenuText     = QT_TR_NOOP("Annotation Label");
+    sToolTipText  = QT_TR_NOOP("Creates a new annotation label at the selected element in the 3D view");
+    sWhatsThis    = "Std_AnnotationLabel";
+    sStatusTip    = sToolTipText;
+    sPixmap       = "Tree_Annotation";
+    eType         = AlterDoc;
+}
+
+bool StdCmdAnnotationLabel::isActive()
+{
+    return (App::GetApplication().getActiveDocument() != nullptr
+            && !Gui::Selection().getSelectionEx().empty());
+}
+
+void StdCmdAnnotationLabel::activated(int)
+{
+    App::Document* doc = App::GetApplication().getActiveDocument();
+    if (!doc) {
+        return;
+    }
+
+    auto selEx = Gui::Selection().getSelectionEx();
+    if (selEx.empty()) {
+        return;
+    }
+
+    // Use first selected item in active 3D view
+    const Gui::SelectionObject& sel = selEx.front();
+
+    const App::DocumentObject* support = sel.getObject();
+    App::DocumentObject* supportObj = const_cast<App::DocumentObject*>(support);
+
+    // Compute base position. Fallback: object placement position, else origin
+    Base::Vector3d basePos;
+    const auto& picked = sel.getPickedPoints();
+    if (!picked.empty()) {
+        basePos = picked.front();
+    } else if (supportObj && supportObj->isDerivedFrom(App::GeoFeature::getClassTypeId())) {
+            auto* gf = static_cast<App::GeoFeature*>(supportObj);
+            const Base::Placement& gp = gf->globalPlacement();
+            basePos = gp.getPosition();

```

Comment:
1. indentation
2. if you see `isDerivedFrom` and `static_cast` pattern it should be replaced with `freecad_cast`

```suggestion
    } else if (auto* gf = freecad_cast<App::GeoFeature*>(supportObj); supportObj && gf) {
        const Base::Placement& gp = gf->globalPlacement();
        basePos = gp.getPosition();
```